### PR TITLE
[host] Add owner blob support for ISFB TLV entry.

### PR DESCRIPTION
**Pull request by moidx**
This change adds support for the Integration Specific Firmware Binding (ISFB) configuration, which describes the configuration parameters of the ISFB region.

This owner blob TLV entry is optional and not required for regular ownership operation.

This is part of: https://github.com/lowRISC/opentitan/issues/24666

### CODE REVIEW

**Review by timothytrippel — APPROVED** [review]

File: sw/host/opentitanlib/src/ownership/isfb.rs [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        skip_serializing_if = "GlobalFlags::not_debug",
+        default = "OwnerIsfbConfig::default_header"
+    )]
+    pub header: TlvHeader,
+    /// The flash bank where the ISFB is located.
+    pub bank: u8,
+    /// The info flash page where the ISFB is located.
+    pub page: u8,
+    /// Padding for alignment.
+    #[serde(default)]
```
Comment (cfrantz) [inline sw/host/opentitanlib/src/ownership/isfb.rs:29]: `#[serde(default, skip_serializing_if = "GlobalFlags::not_debug")]`

File: sw/host/opentitanlib/src/ownership/isfb.rs [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    #[annotate(format=hex)]
+    pub pad: u16,
+
+    /// The erase policy for the ISFB region.
+    #[annotate(format=hex)]
+    pub erase_conditions: u32,
+    /// The key domain associated with the erase condition policy.
+    #[annotate(format=hex)]
+    pub key_domain: u32,
```
Comment (cfrantz) [inline sw/host/opentitanlib/src/ownership/isfb.rs:38]: `pub key_domain: ApplicationKeyDomain`

There are pre-defined constants for `Prod`, `Dev` and `Test`.  The `ApplicationKeyDomain` type also permits arbitrary integer values in addition to the well-known enumerated values.  See `ownership/application_key.rs`.

File: sw/host/opentitanlib/src/ownership/isfb.rs (hunk 1 above)
Comment (moidx) [reply sw/host/opentitanlib/src/ownership/isfb.rs]: Fixed,

File: sw/host/opentitanlib/src/ownership/isfb.rs (hunk 2 above)
Comment (moidx) [reply sw/host/opentitanlib/src/ownership/isfb.rs]: Thanks. Updated.

**Review by cfrantz — APPROVED** [review]

**Pull request merged**